util/src/main/java/com/spotify/hamcrest/util/DescriptionUtils.java

```diff
@@ -77,18 +77,18 @@ public static void indentDescription(Description description, Description innerD
    *                       which will write the describe the mismatch for that key
    * @param describeKey A {@link BiConsumer} used to describe the key
    */
-  public static void describeNestedMismatches(
-      Set<String> allKeys,
+  public static <T> void describeNestedMismatches(
```

Comment:
Ok, I buy it. I don't love using Maps without hashCode, but in this case it will work.